### PR TITLE
Adding status code to web_exception

### PR DESCRIPTION
**Pull request by moozzyk**

### CODE REVIEW

File: include/signalrclient/web_exception.h [hunk 1]
```diff
@@ -12,8 +12,16 @@ namespace signalr
     class web_exception : public std::runtime_error
     {
     public:
-        explicit web_exception(const utility::string_t &what)
-            : runtime_error(utility::conversions::to_utf8string(what))
+        web_exception(const utility::string_t &what, unsigned short status_code)
```
Comment (BrennanConroy) [inline include/signalrclient/web_exception.h:15]: I'm assuming we don't have the same problem with public api changes as the other projects?

Comment (moozzyk) [reply include/signalrclient/web_exception.h:15]: Nope - this is still in Beta. Also users are not expected to create exceptions of this type.

**Comment by davidfowl** [comment]
We flow the exception out to of the client?

**Comment by moozzyk** [comment]
We do when starting the connection (e.g. `negotiate` request fails)

**Comment by jordanrudd** [comment]
Looks good!

**Pull request merged**